packages/itwin/tree-widget/src/components/trees/models-tree/internal/VisibilityChangeEventListener.ts

```diff
@@ -46,5 +49,14 @@ export function createVisibilityChangeEventListener(viewport: Viewport): IVisibi
       listeners.forEach((x) => x());
       listeners.length = 0;
     },
+    suppressChangeEvents: () => {
+      suppressChangeEvents++;
+    },
+    unSuppressChangeEvents: () => {
+      suppressChangeEvents--;
+      if (suppressChangeEvents === 0) {
+        handleVisibilityChange();
```

Comment:
I don't think there is need to add additional logic to track if visibility change events happened while event listener was suppressed because we suppress and resume event listening only when tree visibility handler is changing visibility so I think we can assume that it will make visibility changes.